src/gt4py/next/constructors.py

```diff
@@ -345,7 +345,7 @@ def as_connectivity(
     if shape != actual_domain.shape:
         raise ValueError(f"Cannot construct 'Field' from array of shape '{shape}'.")
     if dtype is None:
-        dtype = storage_utils.asarray(data).dtype
+        dtype = core_defs.dtype(storage_utils.asarray(data).dtype)
```

Comment:
@havogt / @egparedes Ruff got mad because `dtype` != `DType`. I think this is what you want here? Triple check me.

Reply:
see my other comment for the proper fix

Reply:
`mypy` actually

Reply:
All fixed